scripts/update-internal-dependencies.js

```diff
@@ -0,0 +1,75 @@
+const fs = require("fs").promises;
+
+const semver = require("semver");
+
+// get cli arguments. Any cli argument is assumbed to be the name of a package
+// dependency to update in other packages. If no extra cli arguments are
+// provided, then all packages will be updated.
+//
+// For example:
+//   `yarn update-internal-dependencies`
+//      updates everything
+//   `yarn update-internal-dependencies core-utils base-map`
+//      updates packages that depend on either @opentripplanner/core-utils or
+//      @opentripplanner/base-map
+const filteredPackages = process.argv.slice(2);
+if (filteredPackages.length === 0) {
+  console.log("Updating all internal @opentripplanner dependencies!");
+}
+
+function shouldUpdatePackage(pkg) {
+  if (filteredPackages.length === 0) return true;
+  return filteredPackages.indexOf(pkg) > -1;
+}
+
+/**
+ * Updates all internal @opentripplanner/* dependencies in each package's
+ * package.json to the latest version.
+ */
+async function main() {
+  // iterate through all packages to build up dependency tree
+  const packages = await fs.readdir("./packages");
+  const pkgLookup = {};
+  await Promise.all(
+    packages.map(async pkg => {
+      // skip paths that aren't directories
+      if (!(await fs.stat(`./packages/${pkg}`)).isDirectory()) return;
```

Comment:
Minor semantic thing (non-blocking), but why not nest the pkgLookup code below in a check for isDirectory, rather than skipping non-directories? For some reason, doing it this way seems clearer.